dipy/reconst/dki.py

```diff
@@ -0,0 +1,1606 @@
+        
+    .. math::
+            
+    \begin{matrix} ( & W_{xxxx} & W_{yyyy} & W_{zzzz} & W_{xxxy} & W_{xxxz}
+                     & ... \\
+                     & W_{xyyy} & W_{yyyz} & W_{xzzz} & W_{yzzz} & W_{xxyy}
+                     & ... \\
+                     & W_{xxzz} & W_{yyzz} & W_{xxyz} & W_{xyyz} & W_{xyzz}
+                     & & )\end{matrix}
+
+    References
+    ----------
+    [1] Hui ES, Cheung MM, Qi L, Wu EX, 2008. Towards better MR
+    characterization of neural tissues using directional diffusion kurtosis
+    analysis. Neuroimage 42(1): 122-34
+    """
+    if inds is None:
+        inds = np.array([[0, 0, 0, 0], [1, 1, 1, 1], [2, 2, 2, 2],
+                         [0, 0, 0, 1], [0, 0, 0, 2], [0, 1, 1, 1],
+                         [1, 1, 1, 2], [0, 2, 2, 2], [1, 2, 2, 2],
+                         [0, 0, 1, 1], [0, 0, 2, 2], [1, 1, 2, 2],
+                         [0, 0, 1, 2], [0, 1, 1, 2], [0, 1, 2, 2]])
+    else:
+        inds = np.array(inds)
+        inds = inds.reshape((-1, inds.shape[-1]))
+
+    Wrot = np.zeros(len(inds))
+
+    # Construct full 4D tensor
+    W4D = Wcons(kt)
+
+    for e in range(len(inds)):
+        Wrot[e] = _Wrotate_element(W4D, inds[e][0], inds[e][1], inds[e][2],
+                                   inds[e][3], Basis)
+
+    return Wrot
+
+
+def _Wrotate_element(W4D, indi, indj, indk, indl, B):
+    r"""
+    Helper function that returns the element with specified index of a rotated
+    kurtosis tensor from the Cartesian coordinate system to another coordinate
+    system basis
+    
+    Parameters
+    ----------
+    W4D : array(4,4,4,4)
+        Full 4D kutosis tensor in the Cartesian coordinate system
+    indi : int
+        Rotated kurtosis tensor element index i (0 for x, 1 for y, 2 for z)
+    indj : int
+        Rotated kurtosis tensor element index j (0 for x, 1 for y, 2 for z)
+    indk : int
+        Rotated kurtosis tensor element index k (0 for x, 1 for y, 2 for z)
+    indl: int
+        Rotated kurtosis tensor element index l (0 for x, 1 for y, 2 for z)
+    B: array (3, 3)
+        Vectors of the basis column-wise oriented
+    
+    Returns
+    -------
+    Wre : float
+          rotated kurtosis tensor element of index ind_i, ind_j, ind_k, ind_l
+    
+    References
+    ----------
+    [1] Hui ES, Cheung MM, Qi L, Wu EX, 2008. Towards better MR
+    characterization of neural tissues using directional diffusion kurtosis
+    analysis. Neuroimage 42(1): 122-34
+    """
+
+    Wre = 0
+
+    # These for loops can be avoid using kt symmetry properties. If this
+    # simplification is done we don't need also to reconstruct the full kt
+    # tensor
+    for il in range(3):
+        for jl in range(3):
+            for kl in range(3):
+                for ll in range(3):
+                    multiplyB = B[il][indi]*B[jl][indj]*B[kl][indk]*B[ll][indl]
+                    Wre = Wre + multiplyB * W4D[il][jl][kl][ll]
```

Comment:
numpy indexing is done by, `W4D[il, jl, kl, ll]`. The other way works sometimes, but can cause bugs.